src/engine/Operation.cpp

```diff
@@ -80,7 +91,14 @@ shared_ptr<const ResultTable> Operation::getResult(bool isRoot) {
     // Passing the raw pointer here is ok as the result shared_ptr remains
     // in scope
     try {
+      if (_timeoutTimer->wlock()->isTimeout()) {
+        throw ad_semsearch::TimeoutException("uncaught Timeout before " +
+                                             getDescriptor());
+      }
       computeResult(newResult->_resTable.get());
+      if (_timeoutTimer->wlock()->isTimeout()) {
+        throw ad_semsearch::TimeoutException("Timeout in " + getDescriptor());
```

Comment:
Should be "uncaught after" or something like this." (Better names should be integrated here)